dashboard/app/models/census/census_summary.rb

```diff
@@ -28,4 +28,161 @@ class Census::CensusSummary < ApplicationRecord
   enum teaches_cs: TEACHES
 
   validates_presence_of :audit_data
+
+  def self.submission_teaches_cs(submission, is_high_school)
+    if is_high_school
+      (submission.how_many_20_hours_some? || submission.how_many_20_hours_all?)
+    else
+      (
+        submission.how_many_10_hours_some? ||
+        submission.how_many_10_hours_all? ||
+        submission.how_many_20_hours_some? ||
+        submission.how_many_20_hours_all?
+      )
+    end
+  end
+
+  def self.summarize_census_data
+    latest_survey_year = Census::CensusSubmission.maximum(:school_year)
+    latest_ap_data_year = Census::ApCsOffering.maximum(:school_year)
+    latest_ib_data_year = Census::IbCsOffering.maximum(:school_year)
+    latest_state_data_years = {}
+    state_years_with_data = {}
+    Census::StateCsOffering::SUPPORTED_STATES.each do |state|
+      state_years_with_data[state] = Census::StateCsOffering.
+                                       joins(:school).
+                                       where(schools: {state: state}).
+                                       select(:school_year).
+                                       group(:school_year).
+                                       map(&:school_year)
```

Comment:
Does `pluck(:school_year)` work here instead of map? Since we're only selecting that column it probably makes little to no difference so not a big deal.

Reply:
Both work and generate the exact same SQL.